### PR TITLE
Update bulletproofs dep

### PR DESCRIPTION
**Pull request by drewstone**

### CODE REVIEW

File: src/smt/smt.rs [hunk 1]
```diff
@@ -18,8 +18,9 @@ use bulletproofs::{
 	BulletproofGens,
 };
 use curve25519_dalek::{ristretto::CompressedRistretto, scalar::Scalar};
-use rand_core::OsRng;
```
Comment (shekohex) [inline src/smt/smt.rs:21]: we should add a `#[cfg(feature = "std")]` here.

File: src/smt/smt.rs [hunk 2]
```diff
@@ -204,7 +205,7 @@ impl VanillaSparseMerkleTree {
 			Vec<CompressedRistretto>,
 		),
 	) {
-		let mut test_rng: OsRng = OsRng::default();
+		let mut test_rng = ChaChaRng::from_seed([1u8; 32]);
```
Comment (shekohex) [inline src/smt/smt.rs]: this should be `OsRng::default()` with `#[cfg(feature = "std")]` on the `prove_zk`

**Review by shekohex — CHANGES_REQUESTED** [review]

File: src/smt/smt.rs (hunk 2 above)
Comment (drewstone) [reply src/smt/smt.rs]: Updated

File: src/smt/smt.rs (hunk 1 above)
Comment (drewstone) [reply src/smt/smt.rs:21]: Updated

**Pull request merged**